### ISSUE TITLE
Error R6034

### ISSUE THREAD

**Issue by violleTolorza** [dtarb/TauDEM#130]
Dear all,
I'm trying to use TauDEM before begin with other tools, as SedinConnect. I'm using arcGIS 10.2.1 over windows 10. I installed all the tools indicated in the download page, listed here:
    GDAL 201 for 64 bit Windows PC
    GDAL 2.1.0 for Python 2.7
    Microsoft MPI v7.1
    Microsoft Visual C++ 2015 Redistributable Package (x86)
    Microsoft Visual C++ 2015 Redistributable Package (x64)
I installed Microsoft MPI v7.1 by running as administrator from the contextual menu.
Following the advice of SedInConnect people (https://github.com/HydrogeomorphologyTools/SedInConnect_2.3/blob/master/Guidelines_tools_stand-alone.docx), I turned the UAC (User Account Control) to the lowest value to prevent Windows from blocking MPI activity.
I installed TauDEM 5.3.7 by running as administrator from the contextual menu.
I installed the arcGIS toolbox and tried to run pitremove from there. 
I obtain the error R6034:
![dcaptura de pantalla 2017-05-22 12 36 48](https://cloud.githubusercontent.com/assets/28870108/26319646/d6b186ae-3eed-11e7-9c58-f9b0633441cc.png)

 And the ERROR 000732 in the arcGIS pit remove dialog, who cannot calculate statistics in the file I put as output.
I tried to run from the command line, both by using the command prompt and my GIT bash:
mpiexec -n 8 pitremove -z "./FromENVI/S36W072_73_DEM_RBB_UTM18S.tif" -fel "./FromTauDEM/RBB.tif"

And I obtained the same error R6034.

I don't know what to do. I'll appreciate your help,
Violeta

**Comment by dtarb** [comment]
The fact that you get an error on the command line means you can rule out it being a Python or ArcGIS problem.  Did you try run the complete windows installer from http://hydrology.usu.edu/taudem/taudem5/downloads2.html.  This is generally more reliable than installing each component separately, and it also sets up the paths that you need.  The first thing I would suggest is uninstall each of the components you mentioned, then run the complete windows installer.  If this does not work you'll need to start looking into path issues.  If you type "Path" at a command prompt you should see entries
PATH=C:\Program Files\Microsoft MPI\Bin\;C:\Program Files\TauDEM\TauDEM5Exe; C:\Program Files\GDAL;C:\GDAL that TauDEM uses.  

I have never has success running TauDEM from a bash shell, but the regular Windows command prompt works.

**Comment by violleTolorza** [comment]
I uninstall and reinstall. It didn't work.
![captura de pantalla 2017-05-22 15 42 34](https://cloud.githubusercontent.com/assets/28870108/26325608/94731cfe-3f05-11e7-96f2-beb46416988d.png)

In my path I found all the path you mention.
This is my path:
PATH=C:\Program Files\Microsoft MPI\Bin\;C:\ProgramData\Oracle\Java\javapath;C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Anaconda2;C:\Anaconda2\Scripts;C:\Anaconda2\Library\bin;C:\Program Files (x86)\PDFtk\bin\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\Windows Live\Shared;C:\GDAL;C:\Program Files\GDAL;C:\Program Files\TauDEM\TauDEM5Exe;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\Program Files (x86)\Skype\Phone\;C:\Program Files\MiKTeX 2.9\miktex\bin\x64\;C:\Users\violl\AppData\Local\Microsoft\WindowsApps;;C:\Program Files (x86)\SSH Communications Security\SSH Secure Shell

**Comment by dtarb** [comment]
I see that Anaconda is installed.  I have had trouble when I have installed Anaconda then tried to use TauDEM.  I have not got to the source of the problem, but I think it may be that Anaconda installs a different set of Microsoft Redistributable libraries.  I recognize that it is a drastic ask to suggest removing Anaconda, but I would try a few things incrementally first.
1.  Move C:\GDAL;C:\Program Files\GDAL;C:\Program Files\TauDEM\TauDEM5Exe; to the beginning of your path and try
2.  Remove Anaconda entries from the path and try.
3.  Remove Anaconda
I fear you may need to go all the way to step 3.  My theory is that Anaconda uses different Microsoft run time libraries Microsoft Visual C++ 2015 Redistributable Package (x86) and Microsoft Visual C++ 2015 Redistributable Package (x64) and these are installed in a system folder somewhere so fiddling with the path does not fix.

**Comment by violleTolorza** [comment]
I couldn't change the path...it seems to be a problem of win 10. I've administrator rights, but I couldn't change the path with any of the methods proposed here: https://superuser.com/questions/949560/how-do-i-set-system-environment-variables-in-windows-10 (including SETX), so I'll uninstall anaconda and I'll tell you what happen. 
Thanks very much

**Comment by dtarb** [comment]
Here is the path editing sequence I use
![image](https://cloud.githubusercontent.com/assets/3456833/26331486/61534f34-3f0e-11e7-8613-d7516d7dde39.png)

**Comment by dtarb** [comment]
Here is the path editing sequence I use
![image](https://cloud.githubusercontent.com/assets/3456833/26331504/7a7c5e60-3f0e-11e7-8336-bc889db36b37.png)

**Comment by violleTolorza** [comment]
I used the same path editing sequence, but after accepting, closing and re-opening the path, the changes were not saved. I notice also I cannot change the system path from the user account item:
![inkedcaptura de pantalla 2017-05-23 12 25 18_li](https://cloud.githubusercontent.com/assets/28870108/26370105/cdf6d654-3fc4-11e7-9e80-f68b494284f0.jpg)
I uninstalled Anaconda and Taudem works properly. However, I'll need Anaconda in the following weeks. When I reinstalled Anaconda I'll come back to the "PATH" issue.
Thanks very much
